### PR TITLE
D8AGE-871: Fix the callback URLs in the translation request.

### PR DESCRIPTION
**Pull request by intelektron**
### Description

Right now, the callbacks in the `modules/oe_translation_cdt/src/Mapper/TranslationRequestMapper.php:173` file are assigned to wrong routes. The `oe_translation_cdt.request_status_callback` should be assigned to `$request_status_url`

### CODE REVIEW

File: modules/oe_translation_cdt/modules/oe_translation_cdt_mock/oe_translation_cdt_mock.install [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (intelektron) [inline modules/oe_translation_cdt/modules/oe_translation_cdt_mock/oe_translation_cdt_mock.install:1]: I also added this change to make testing a bit more reliable, previously the mock module would throw an error if the old token was still valid.

File: modules/oe_translation_cdt/src/Mapper/TranslationRequestMapper.php [hunk 2]
```diff
@@ -175,14 +175,14 @@ protected function countVolume(int $character_count): float {
   protected function createCallbacks(): CallbackCollection {
     // Set the LANGUAGE_NOT_SPECIFIED to avoid the language suffix in the URL.
     // @see \Drupal\language\Plugin\LanguageNegotiation\LanguageNegotiationUrl::processOutbound()
-    $job_status_url = Url::fromRoute(
+    $request_status_url = Url::fromRoute(
```
Comment (drishu) [inline modules/oe_translation_cdt/src/Mapper/TranslationRequestMapper.php:178]: Cant belive we missed it, lets improve also in TranslationRequestMapperTest the testing of getCallbackBaseUrl

**Review by drishu — CHANGES_REQUESTED** [review]
Not sure since when, but drone seems to fail in cloning the cdt client

File: modules/oe_translation_cdt/src/Mapper/TranslationRequestMapper.php (hunk 2 above)
Comment (intelektron) [reply modules/oe_translation_cdt/src/Mapper/TranslationRequestMapper.php:178]: True! I've just added some additional asserts.

**Comment by intelektron** [comment]
I rebased the epic onto latest 2.x, and fixed the remaining GrumPHP issues. We will be deploying the module with R29 anyway, and by doing this we get rid of Drone problems.

**Review by drishu — APPROVED** [review]

**Pull request merged**